### ISSUE TITLE
SemVersion.objects.get_or_create invalid behaviour

### ISSUE THREAD

**Issue by ejosan** [rbarrois/python-semanticversion#97]
###problem
running get_or_create twice will result in the first run to fail and the second to pass

### the model
from django.db.models import Model
from semantic_version.django_fields import VersionField
class SemVersion(Model):
    version = VersionField(max_length=200)

####test script
# -*- coding: utf-8 -*-
# from __future__ import absolute_import, unicode_literals, print_function
import unittest
import django
django.setup()
from apps.version.models import SemVersion

class SemverTest(unittest.TestCase):

    def test_one(self):
        obj1,_ = SemVersion.objects.get_or_create(version='0.1.1+build')
        print(obj1)
        print(obj1.version.major)
        print(obj1.version.minor)
        print(obj1.version.patch)
        print(obj1.version.prerelease)
        print(obj1.version.build)

if __name__ == '__main__':
    suite = unittest.TestSuite()
    suite.addTest(SemverTest('test_one'))

    result = unittest.TextTestRunner(verbosity=2, resultclass=unittest.TextTestResult).run(suite)

### output first run:
test_one (__main__.SemverTest) ... SemVersion object
ERROR

======================================================================
ERROR: test_one (__main__.SemverTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ONT/user/apps/version/tests/test.py", line 20, in test_one
    print(obj1.version.major)
AttributeError: 'str' object has no attribute 'major'

----------------------------------------------------------------------
Ran 1 test in 0.022s

FAILED (errors=1)

### output second run:
test_one (__main__.SemverTest) ... SemVersion object
0
1
1
()
('build',)
ok

----------------------------------------------------------------------
Ran 1 test in 0.015s

OK

looks like the the create part of get_or_create returns not an object but a string, the second time get_or_create is run the get part kicks in and returns a correct version object. This is inconsistent and should work. Tested it with python 3.6 and 2.7 both same results. Django 1.11.15

**Comment by rbarrois** [comment]
Thanks for the report! That's indeed a bug, likely because Django's `get_or_create` returns the value as is.

**Comment by rbarrois** [comment]
After taking a deeper look, it seems to be a standard behaviour from Django: if the developer provides a badly-typed-but-coercable value, it might be used properly by the database, but the value on the model won't be changed.

This is, unfortunately, not properly documented in Django.

Here are a couple of links on that topic:
- An issue regarding `DecimalField`: https://code.djangoproject.com/ticket/27825;
- A discussion on the django-developers mailing list: https://groups.google.com/forum/#!msg/django-developers/3IX1zH09Idg/s7mSmWZhBgAJ.

Basically, a `VersionField` shouldn't accept a string, only a `Version` object; that's the bug ;)

**Comment by ejosan** [comment]
> After taking a deeper look, it seems to be a standard behaviour from Django: if the developer provides a badly-typed-but-coercable value, it might be used properly by the database, but the value on the model won't be changed.
> 
> This is, unfortunately, not properly documented in Django.
> 
> Here are a couple of links on that topic:
> 
> * An issue regarding `DecimalField`: https://code.djangoproject.com/ticket/27825;
> * A discussion on the django-developers mailing list: https://groups.google.com/forum/#!msg/django-developers/3IX1zH09Idg/s7mSmWZhBgAJ.
> 
> Basically, a `VersionField` shouldn't accept a string, only a `Version` object; that's the bug ;)

;-)
